src/auth.ts

```diff
@@ -79,32 +88,32 @@ export const hasFileAccess = (
       fileNames: [],
     };
   }
-  // Evaluate access rights based on institution specific logic
-  switch (facility) {
-  case "maxiv": {
-    return authMAXIV(authRequest);
-  }
-  default:
-    return {
-      hasAccess: true,
-      statusCode: 200,
-      directory: authRequest.directory,
-      fileNames: authRequest.fileNames,
-    };
-  }
-};
 
-const authMAXIV = (authRequest: Global.AuthRequest): Global.AuthResponse => {
-  const valid =
-    authRequest.jwt.groups.filter(
-      (group) => group.trim() && authRequest.directory.indexOf(group) > -1
-    ).length > 0;
 
-  return {
-    hasAccess: valid,
-    statusCode: valid ? 200 : 403,
-    error: valid ? "" : "You do not have access to this resource",
-    directory: valid ? authRequest.directory : undefined,
-    fileNames: valid ? authRequest.fileNames : [],
-  };
-};
+
+  const valid = await dataSetAPI.datasetsControllerFindById({pid: authRequest.dataset}).then(
+    (value) => 
+      {
+        if(value.isPublished ||  // Check if proposal is public
+          value.accessGroups.some(item => new Set(authRequest.jwt.groups).has(item)) ||  // Check if user has one or more of the access groups of dataset
+          authRequest.jwt.groups.indexOf(value.ownerGroup) > -1) //Check if user has the owner group
+          {
+          return true;
+        }
```

Comment:
```suggestion
const isPublic = value.isPublished; 
const hasAccessGroup = value.accessGroups.some(item => new Set(authRequest.jwt.groups).has(item)); 
const isOwnerGroup = authRequest.jwt.groups.includes(value.ownerGroup);  
if (isPublic || hasAccessGroup || isOwnerGroup) {
  return true;
}
```
It'd be easier to read this way, what do you think?

Reply:
I agree

Reply:
The suggested codes should be inside the curly bracket.